docs/topics/proc_cli-execute.adoc

```diff
@@ -25,56 +27,115 @@ $ ./mta-cli --input /path/to/jee-example-app-1.0.0.ear --output /path/to/output
 
-[source,options="nowrap",subs="+quotes"]
+[source,terminal]
 ----
-$ <MTA_HOME>/bin/mta-cli --input /path/to/jee-example-app-1.0.0.ear --output /path/to/report-output/ --target eap:7 --target cloud-readiness --packages com.acme org.apache
+$ <MTA_HOME>/bin/mta-cli --input /path/to/jee-example-app-1.0.0.ear \
+ --output /path/to/report-output/ \
+ --target eap:7 --target cloud-readiness --packages com.acme org.apache
 ----
 
 [discrete]
 === Overriding {ProductShortName} properties
 
 To override the default _Fernflower_ decompiler, pass the `-Dwindup.decompiler` argument on the command line. For example, to use the _Procyon_ decompiler, use the following syntax:
 
-[source,options="nowrap",subs="+quotes"]
+[source,terminal]
 ----
-$ <MTA_HOME>/bin/mta-cli -Dwindup.decompiler=procyon --input <INPUT_ARCHIVE_OR_DIRECTORY> --output <OUTPUT_REPORT_DIRECTORY> --target <TARGET_TECHNOLOGY> --packages <PACKAGE_1> <PACKAGE_2>
+$ <MTA_HOME>/bin/mta-cli -Dwindup.decompiler=procyon --input \
+<INPUT_ARCHIVE_OR_DIRECTORY> --output <OUTPUT_REPORT_DIRECTORY> \
+--target <TARGET_TECHNOLOGY> --packages <PACKAGE_1> <PACKAGE_2>
 ----
+[id="openrewrite_{context}"]
+== Refactoring source code using OpenRewrite
+
+link:https://docs.openrewrite.org/[OpenRewrite] uses recipes to automate large-scale, distributed source code refactoring. It can be used with the {ProductShortName}.
+
+The first OpenRewrite recipe, which is shipped with {ProductShortName} 5.2.1, renames imported `javax` packages to their `jakarta` equivalents.
+
+You can use OpenRewrite in the {ProductShortName} CLI to prepare Java applications for migration.
+
+.Procedure
+
+. Add the `openrewrite` argument when you execute the 'mta-cli' script and specify the other arguments appropriately:
++
+[source,terminal]
+----
+./mta-cli --openrewrite
+"-DactiveRecipes=<recipe name>" --input
+/path/to/source/project --goal dryRun
+----
+
+* `--openrewrite`: Flag specifying to run an OpenRewrite migration, instead of an MTA analysis.
+
+* `"-DactiveRecipes=<recipe name>"` : The OpenRewrite recipe to apply to the input project. `org.jboss.windup.JavaxtoJakarta` is the default shipped recipe but you can add your own to the shipped `rewrite.yml` and run that recipe instead. The `rewrite.yml file` is located in the `rules/openrewrite/`` folder in the uncompressed MTA distribution.
+
+* `--input`: The application to be evaluated.
+
+* `--goal`: Optional. The OpenRewrite maven goal to run. Parameters:
+** `dryRun` : The script returns a list of proposed changes, creates a patch file which can be applied later, and returns the following message:
++
+[source,terminal]
+----
+Run 'mvn rewrite:run' to apply the recipes.
+----
+** `run`: The script makes all changes automatically without listing them first.
++
+If you do not enter a parameter for `goal`, the script executes it as `dryRun`.
```

Comment:
I do not think you need any of the "goal" documentation. It looks like it is self-documenting. 

If you have to document it, I would add "allowed values are 'dryRun', to see a list of proposed changes, or 'run', to apply the changes.